server/src/server.js

```diff
@@ -585,7 +588,11 @@ app.put("/data/:id/:domain", function(req, res) {
     res.end();
     return;
   }
-  let bodyObj = req.body;
+  let bodyObj = JSON.parse(req.body.shot);
+  let clipId = Object.getOwnPropertyNames(bodyObj.clips)[0];
+  let b64 = req.file.buffer.toString("base64");
+  b64 = "data:image/png;base64," + b64;
+  bodyObj.clips[clipId].image.url = b64;
```

Comment:
This is where it breaks. Probably `req.file` won't be defined if there's no multipart upload? In which case the image should have a clip image already.